### PR TITLE
add sbt 2.x build setting

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -61,9 +61,11 @@ val sbtdynver = project.dependsOn(dynverLib).enablePlugins(SbtPlugin).settings(
   scriptedDependencies := Seq(dynver / publishLocal, publishLocal).dependOn.value,
   scriptedLaunchOpts   += s"-Dplugin.version=${version.value}",
   scriptedLaunchOpts   += s"-Dsbt.boot.directory=${file(sys.props("user.home")) / ".sbt" / "boot"}",
+  crossScalaVersions   += scala3,
   (pluginCrossBuild / sbtVersion) := {
     scalaBinaryVersion.value match {
       case "2.12" => "1.3.0"
+      case _ => "2.0.0-M2"
```
Comment (eed3si9n) [inline build.sbt:68]: Could you update the CI script to run `+scripted` plz?

**Comment by xuwei-k** [comment]
```
[error] Expected ';'
[error] No such setting/task
[error] ++  2.13.15 {file:/home/runner/work/sbt-dynver/sbt-dynver/}dynver/scripted
[error]                                                                           ^
```

🤔

**Comment by eed3si9n** [comment]
It's cross building the plugin using Scala 2.13. I wonder if you should change to:

```scala
crossScalaVersions := Nil
```

at line 3, and set `crossScalaVersions` with `:=` in each subproject?

**Comment by xuwei-k** [comment]
Maybe

```scala
// no `scripted` task
val p1 = project

// has `scripted` task
val p2 = project.enablePlugins(SbtPlugin)
```

`sbt scripted` success.
but `sbt +scripted` failure. need `+ p2/scripted`

**Comment by xuwei-k** [comment]
- https://github.com/sbt/sbt-dynver/pull/306

**Comment by eed3si9n** [comment]
I got https://github.com/sbt/sbt-dynver/pull/314 to work after many trial and errors.

**Pull request closed without merging**